### PR TITLE
Gslux 623 3dlayers

### PR DESCRIPTION
**Pull request by mki-c2c**
### JIRA issue

https://jira.camptocamp.com/browse/GSLUX-623

### Description

all features for 3d layers united

### Screenshots

(only if the final render is different)

### CODE REVIEW

File: src/components/map/map-container.vue [hunk 1]
```diff
@@ -44,6 +54,7 @@ provide('olMap', olMap)
     <zoom-to-extent-control :extent="DEFAULT_EXTENT" />
     <fullscreen-control />
     <attribution-control />
+    <map-3d-control v-if="v4_standalone" />
```
Comment (AlitaBernachot) [inline src/components/map/map-container.vue:57]: Not a big fan of this, but I understand the need.
Maybe we should handle this with vuejs slots? But lets keep this as is for now.

File: src/composables/layers/layers.composable.ts [hunk 2]
```diff
@@ -14,13 +14,9 @@ const themes = useThemes()
...
     try {
-      const concat: number[] = JSON.parse(exclusionA)
-        .concat(JSON.parse(exclusionB))
-        .sort((a: number, b: number) => a - b)
+      const concat = JSON.parse(exclusionA).concat(JSON.parse(exclusionB))
 
-      return concat.some(
-        (element, index, array) => index && element === array[index - 1]
-      )
+      return new Set(concat).size < concat.length
```
Comment (AlitaBernachot) [inline src/composables/layers/layers.composable.ts:19]: I don't get how comparing the size of a set can detect an exclusion, what if there is the same length on both layers but not the same ids?

**Review by AlitaBernachot — APPROVED** [review]
Thanks for the work!

File: src/components/layer-manager/layer-manager.vue [hunk 3]
```diff
@@ -22,7 +22,7 @@ const mapStore = useMapStore()
 const appStore = useAppStore()
 const styles = useMvtStyles()
 const sliderStore = useSliderComparatorStore()
-const { bgLayer } = storeToRefs(mapStore)
+const { layers_3d, bgLayer } = storeToRefs(mapStore)
 const { sliderActive } = storeToRefs(sliderStore)
 
 const layers = computed(() => [...mapStore.layers].reverse())
```
Comment (AlitaBernachot) [inline src/components/layer-manager/layer-manager.vue:28]: There should be a computed const for `layers_3d` (list display in the tree is reversed)

**Comment by AlitaBernachot** [comment]
@tkohr review is done, I let you merge if it is ok for you. Thanks!

File: src/stores/config.store.ts [hunk 4]
```diff
@@ -13,6 +16,20 @@ export const useThemeStore = defineStore(
       themes.value?.find(theme => theme.name === themeName.value)
     )
     const bgLayers = computed(() => config.value?.background_layers || [])
+    const layerTrees_3d = computed(() => {
+      const ol3d_groups = themes.value?.filter(
+        theme => theme.metadata?.ol3d_type !== undefined
```
Comment (AlitaBernachot) [inline src/stores/config.store.ts:21]: Should also test ` if (theme_ol3d_type !== 'terrain')` as in legacy.
Also, there is a layer missing (compared to the prod), entry is missing in fixtures and computed should return this missing layer.
image

Comment (AlitaBernachot) [reply src/stores/config.store.ts:21]: @tkohr FYI update fixtures and translations in https://github.com/Geoportail-Luxembourg/luxembourg-geoportail/pull/80

File: src/composables/layers/layers.composable.ts (hunk 2 above)
Comment (mki-c2c) [reply src/composables/layers/layers.composable.ts:19]: concat is the concatenation of both arrays. If there are ids which are in both arrays, they will be removed by the Set creation, so the size will become smaller

File: src/components/layer-manager/layer-item/layer-item.vue [hunk 5]
```diff
@@ -52,6 +53,7 @@ function changeTime(dateStart?: string, dateEnd?: string) {
   <div class="lux-layer-manager-item relative">
     <div class="w-full flex flex-nowrap items-start gap-x-2">
       <button
+        v-if="!is3d"
         class="fa-solid fa-bars cursor-move mt-1"
         :class="draggableClassName"
```
[the remark this answers is not in the document]
Comment (mki-c2c) [reply src/components/layer-manager/layer-item/layer-item.vue]: draggable is not working on purpose

File: src/components/layer-manager/layer-item/layer-item.vue [hunk 6]
```diff
@@ -52,6 +53,7 @@ function changeTime(dateStart?: string, dateEnd?: string) {
   <div class="lux-layer-manager-item relative">
     <div class="w-full flex flex-nowrap items-start gap-x-2">
       <button
+        v-if="!is3d"
```
[the remark this answers is not in the document]
Comment (mki-c2c) [reply src/components/layer-manager/layer-item/layer-item.vue]: not in prod

**Review by tkohr — APPROVED** [review]
Thanks for this tremendous work @mki-c2c @AlitaBernachot! I'll go for the merge and then we can continue from there.

**Pull request merged**